### ISSUE TITLE
[Feature request] Prevent unknown classes (that follow Tailwind convention)

### ISSUE THREAD

**Issue by nicooprat** [francoismassart/eslint-plugin-tailwindcss#367]
**Is your feature request related to a problem? Please describe.**
Example: in our team, some developers keep confusing `gray` and `grey`.

**Describe the solution you'd like**
Would be cool to report `bg-gray-100` as an error because it follows Tailwind convention but doesn't exist.

**Describe alternatives you've considered**
* We'd rather not duplicate the entries in the theme to prevent confusion & bloat in every tools (like IDE autocomplete, Storybook, ...).
* I thought about using `no-restricted-syntax` ESLint rule but it won't cover every case (Vue classes, `@apply` tags, ...)

**Additional context**

**Comment by kachkaev** [comment]
👋 @nicooprat! You should be able for that [tailwindcss/no-custom-classname](https://github.com/francoismassart/eslint-plugin-tailwindcss/blob/master/docs/rules/no-custom-classname.md) rule for that.

**Comment by blue-infinex** [comment]
I'm currently refactoring a project to use semantic tokens instead of scale values e.g. `text-red-500` → `text-critical`. I think it's a good use case for `blacklist`, or similar.

Unless there's a way to configure this using the existing API?

**Comment by blue-infinex** [comment]
I'm currently refactoring a project to use semantic tokens instead of scale values e.g. `text-red-500` → `text-critical`. I think it's a good use case for `blacklist` or similar.

@nicooprat Unless there's a way to configure this behaviour using the existing API?

**Comment by blue-infinex** [comment]
I'm currently refactoring a project to use semantic tokens instead of scale values e.g. `text-red-500` → `text-critical`. I think it's a good use case for `blacklist` or similar.

@kachkaev Unless there's a way to configure this behaviour using the existing API?

**Comment by kachkaev** [comment]
@blue-infinex I see two options here:

1. You can remove colors like `text-red-500` from your theme. [`tailwindcss/no-custom-classname`](https://github.com/francoismassart/eslint-plugin-tailwindcss/blob/master/docs/rules/no-custom-classname.md) will flag them but also all existing instances will stop working (all replacements will need to happen in one go).

1. You can create a [custom ESLint rule](https://eslint.org/docs/latest/extend/custom-rules) or configure [`no-restricted-syntax`](https://eslint.org/docs/latest/rules/no-restricted-syntax). Simplified example:

    ```js
    {
      selector: `Literal[value=/(bg|border)-(red|green)-500/]`,
      message: "Please use a theme color instead"
    }
    ```

    With this approach, you can start by marking existing cases (`// eslint-disable-next-line -- TODO: fix`) and then gradually fix them when there is time.

**Comment by nicooprat** [comment]
For Vue, this is what Cursor suggested:

```
'vue/no-restricted-syntax': [
  {
    selector:
      'VAttribute[key.name="class"][value.value=/gray/] , VAttribute[directive=true][key.argument.name="class"] > VExpressionContainer:matches([value=/gray/], ArrayExpression > Literal[value=/gray/], ObjectExpression > Property > Literal[value=/gray/])',
    message: 'Use "grey" instead of "gray" because it does not exist in our Tailwind config.',
  }
]
```

<img width="903" alt="Image" src="https://github.com/user-attachments/assets/ddd61ae6-52c9-4b42-a0b9-dd70f9645e95" />

Hope it helps!
